src/auth/AuthContext.ts

```diff
@@ -44,6 +44,19 @@ export class AuthContext {
     return project.members?.map((m) => m.username).includes(this.username) || false;
   }
 
+  async isEventParticipant(eventId: string): Promise<boolean> {
+    const userProjects = await Container.get(PrismaClient).project.findMany({
+      where: {
```

Comment:
You can check eventId in the where, and then do count instead of findMany. Should be faster.